core/src/Streamly/Internal/Data/Unfold.hs

```diff
@@ -507,6 +511,66 @@ postscan (Fold stepF initial extract) (Unfold stepU injectU) =
 
     step Nothing = return Stop
 
+{-# INLINE_NORMAL scanWith #-}
+scanWith :: Monad m => Bool -> Fold m b c -> Unfold m a b -> Unfold m a c
+scanWith restart (Fold stepF initial extract) (Unfold stepU injectU) =
+    Unfold step inject
+
+    where
+
+    inject a =  do
+        r <- initial
+        case r of
+            FL.Partial fs -> do
+                x <- injectU a
+                return (Just (fs, x), Nothing)
```

Comment:
A scan always yields the initial value. This is not correct. Can you write a test for this?